from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("p", double_t, 1, "Proportional gain", 1.0, 0.0, 100.0)
gen.add("i", double_t, 1, "Integral gain", 0.0, 0.0, 100.0)
gen.add("d", double_t, 1, "Derivative gain", 0.0, 0.0, 100.0)
gen.add("i_clamp", double_t, 1, "Integral clamp", 10.0, 0.0, 100.0)
gen.add("antiwindup", bool_t, 1, "Enable anti-windup", True)

exit(gen.generate("hero_chassis_controller", "hero_chassis_controller", "ChassisPidConfig"))
